### PR TITLE
Fix a regression in station_api.

### LINKED ISSUE #407: Frontend broken
Exception in thread Thread-4:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 810, in __bootstrap_inner
    self.run()
  File "openhtf/io/frontend/__init__.py", line 119, in run
    self._test.wait_for_update(timeout_s=self._wait_timeout_s))
  File "openhtf/io/station_api.py", line 283, in wait_for_update
    cached_state, summary_dict = self.cached_state_summary
  File "openhtf/io/station_api.py", line 266, in cached_state_summary
    phases=len(test_record.phases),
UnboundLocalError: local variable 'test_record' referenced before assignment

### PR DESCRIPTION
**Pull request by jettisonjoe**
Fixes #407.

### CODE REVIEW

**Comment by fahhem** [comment]
LGTM

File: openhtf/io/station_api.py [hunk 1]
```diff
@@ -632,10 +632,9 @@ def wait_for_plug(self, test_id, plug_type_name, current_state, timeout_s):
     return test_state.plug_manager.wait_for_plug_update(
         plug_type_name, current_state, timeout_s)
 
-  def _summary_for_state(self, state):
+  @classmethod
```
Comment (grybmadsci) [inline openhtf/io/station_api.py]: staticmethod, yeah? don't think you use cls anywhere

File: openhtf/io/station_api.py [hunk 2]
```diff
@@ -699,7 +698,8 @@ def wait_for_update(self, test_uid, remote_state_dict, timeout_s):
           'RPC:wait_for_update() -> short-circuited wait (local was blank)')
       return self._serialize_state_dict(state._asdict())
 
-    state_dict_summary = self._summary_for_state(state)
+    state_dict, event = state.asdict_with_event()
```
Comment (grybmadsci) [inline openhtf/io/station_api.py]: while you're here, maybe rename `event` to `update_event`, it's a big vague as-is.

**Comment by grybmadsci** [comment]
LGTM, pending couple minor nits.

File: openhtf/io/station_api.py (hunk 1 above)
Comment (jettisonjoe) [inline openhtf/io/station_api.py]: Yep, that's what I meant. Thanks.

File: openhtf/io/station_api.py (hunk 2 above)
Comment (jettisonjoe) [inline openhtf/io/station_api.py]: Done.

**Comment by jettisonjoe** [comment]
Merging. Thanks for the review, folks!

**Pull request merged**